internal/provider/resource_custom_domain.go

```diff
@@ -97,6 +100,16 @@ func (r *CustomDomainResource) Schema(ctx context.Context, req resource.SchemaRe
 				MarkdownDescription: "DNS record value of the custom domain.",
 				Computed:            true,
 			},
+			"target_port": schema.Int64Attribute{
+				MarkdownDescription: "Target port of the service.",
+				Required:            true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+				Validators: []validator.Int64{
+					int64validator.Between(0, 65535),
+				},
```

Comment:
While `targetPort` is not marked as required in Railway's API and it is still possible to create custom domain without port over the API, in fact the domain will not work returning an error every time. 

Effectively `targetPort` is required value in order to make custom domain work

Reply:
This is a bug with the provider that I fixed separately. It was wrongly setting targetPort to be `0` if not provided.

Do you use a specific targetPort when setting custom domain? All my apps use default null for this and thus I didn't find the need to add this prop after I fixed the above bug.